### ISSUE TITLE
Cannot build basement with GHC 9.0: unsafeCoerce# no longer exported by GHC.Prim in base 4.15+

### ISSUE THREAD

**Issue by LaurentRDC** [haskell-foundation/foundation#548]
Hello,

I wanted to bring to your attention that I could not build my [package pandoc-plot](https://github.com/LaurentRDC/pandoc-plot) with GHC 9.0.1 because in basement-0.0.11, `unsafeCoerce#` is imported from `GHC.Prim`:

```haskell
...
basement                         > [16 of 84] Compiling Basement.Floating
basement                         > [17 of 84] Compiling Basement.Monad
basement                         > 
basement                         > /tmp/stack-68bc6b9780c2c4d5/basement-0.0.11/Basement/Monad.hs:98:31: error:
basement                         >     Variable not in scope:
basement                         >       unsafeCoerce#
...
```

From the [migrating guide](https://gitlab.haskell.org/ghc/ghc/-/wikis/migration/9.0#unsafecoerce):

> The unsafeCoerce# function has been moved from GHC.Prim to Unsafe.Coerce. As a result, attempting to import unsafeCoerce# from GHC.Prim (or GHC.Base, which previously re-exported unsafeCoerce#) will result in an error with GHC 9.0. A backwards-compatible way to fix the error is to import unsafeCoerce# from GHC.Exts instead.

**Comment by sjakobi** [comment]
Ping @vincenthz. This is blocking support for GHC 9.0 in a lot of projects.

**Comment by sjakobi** [comment]
Ping @vincenthz! :)

**Comment by sjakobi** [comment]
Ping @vincenthz! :) Is there anything particular that keeps you from addressing this?

**Comment by sjakobi** [comment]
FYI: As a Hackage trustee, I have created a Hackage revision that prevents `basement-0.0.11` from building with GHC 9.0.1: http://hackage.haskell.org/package/basement-0.0.11/revisions/

**Comment by fromjavatohaskell** [comment]
I have found that hoogle also can't be build with currently released basement-0.0.11 under ghc-9.0.1.
I have checked - basement from master seems to be compiling fine with ghc-9.0.1.
It looks like latest commit from 7 hours ago has helped https://github.com/haskell-foundation/foundation/commit/3c195907c7e7c41fa0888237b6e36cdbde67df59
I have created local package for basement from master and was able to move forward in compiling hoogle.
But now I am stuck at compiling foundation with 9.0.1
